## Aufgabe 7.1
Old Faithful ist ein Geysir im Yellowstone National Park in Wyoming:
https://en.wikipedia.org/wiki/Old_Faithful
Im Datensatz oldfaithful.txt sind Längen von 272 aufeinanderfolgenden Ausbrüchen aufgeführt. Hier ist ein Histogramm der Daten.

a) Schätzen Sie den Mittelwert der Eruptionsdauern ab und geben Sie ein 95 % Vertrauensintervall mit dem vereinfachten Bootstrap-Verfahren im Skript an. Verwenden Sie dazu 1000 Bootstraps.
Setzen Sie für ∗ den Pfad, wo sich Ihr o

In [7]:
import numpy as np
x = np.loadtxt("oldfaithful.txt")
print(x.mean())

209.2683823529412


In [5]:
n = np.size(x)
nboot = 1000
tmpdata = np.random.choice(x, n*nboot, replace=True)
bootstrapsample = np.reshape(tmpdata, (n, nboot))
xbarstar = np.mean(bootstrapsample, axis=0)
d = np.percentile(xbarstar, q=[2.5, 97.5])
print("Vertrauensintervall: ",d)

Vertrauensintervall:  [200.71295956 217.43520221]


b) Machen Sie dasselbe für den Median

In [11]:
print(np.median(x))

n = np.size(x)
nboot = 1000
tmpdata = np.random.choice(x, n*nboot, replace=True)
bootstrapsample = np.reshape(tmpdata, (n, nboot))
xbarstar = np.median(bootstrapsample, axis=0)
d = np.percentile(xbarstar, q=[2.5, 97.5])
print("Vertrauensintervall: ",d)

240.0
Vertrauensintervall:  [230.  246.5]


c) Schätzen Sie die Wahrscheinlichkeit P(|x − µ| > 5)
ab.
Ersetzen Sie zur Abschätzung µ durch den Mittelwert der Daten und x durch
die Bootstraps

In [13]:
n = x.size
nboot = 1000
tmpdata = np.random.choice(x, n*nboot, replace=True)

bootstrapsample = np.reshape(tmpdata, (n, nboot))
xbarstar = np.mean(bootstrapsample, axis=0) - np.mean(x)
l = np.sum(xbarstar < -5)
u = np.sum(xbarstar > 5)
ratio = (l+u)/nboot

print(ratio)

0.231


## Aufgabe 7.3
Aufgrund langjähriger Untersuchungen ist bekannt, dass der Bleigehalt X von gewissen Bodenproben annähernd normal-verteilt ist
X ∼ N (µ, σ
2
)


a) Es wurden in 10 Bodenproben der Bleigehalt X gemessen. Dabei wurde ein
Mittelwert von x10 = 31 ppb erhalten. Die Standardabweichung sei bekannt
und beträgt 6 ppb.
Geben Sie das zweiseitige 99 % Vertrauensintervall für den Mittelwert an

In [24]:
from scipy.stats import norm, t
import numpy as np
n = 10
ew = 31
sa = 6
a = 0.99

norm.interval(alpha=a, loc=ew, scale=sa/np.sqrt(n))

(26.112707522188142, 35.887292477811854)

b) Wie viele Beobachtungen sind nötig, um die Breite des in Teilaufgabe a) bestimmten zweiseitigen Vertrauensintervalls auf die Hälfte zu reduzieren?
Wie viele (unabhängige) Bestimmungen des Bleigehalts müssen geplant werden, falls der Bleigehalt mit einer Stichprobe „1 ppb genau” bestimmt werden
soll, d. h., wenn die Breite des 99 % des Konfidenzintervalls nicht grösser als
1 ppb sein soll?

In [25]:
# halbieren des intervalls
i = norm.interval(alpha=a, loc=ew, scale=sa/np.sqrt(n*4))
print(i)
print(i[1] - i[0])

(28.556353761094073, 33.44364623890593)
4.8872924778118545


In [30]:
n = 10
i = 20
while i > 1:
    n=n+1
    i = norm.interval(alpha=a, loc=ew, scale=sa/np.sqrt(n))
    i = i[1] - i[0]
print(n)

956


c) Normalerweise ist die Standardabweichung σ unbekannt. Um welchen Faktor
verändert sich die Breite des zweiseitigen Vertrauensintervalls in Teilaufgabe a),
wenn man die Standardabweichung aus den Daten geschätzt hat, also bσ = 6 ?



In [32]:
n = 10
i = norm.interval(alpha=a, loc=ew, scale=sa/np.sqrt(n))
print(i)
it = t.interval(alpha=a, df=n-1, loc=ew, scale=sa/np.sqrt(n))
print(it)

(26.112707522188142, 35.887292477811854)
(24.833870595963425, 37.166129404036575)


In [34]:
f = (it[1]-it[0]) / (i[1]-i[0])
print(f)

1.2616657243311291


Aufgabe 7.4
Im National Bureau of Standards (USA) wurden regelmässig Wägungen des 10-Gramm
- Standardgewichtstücks durchgeführt. Bei 9 Wägungen erhielt man als durchschnittliche Differenz −403 Mikrogramm vom 10 Gramm-Sollgewicht und eine Standardabweichung von 3.127 Mikrogramm für eine einzelne Wägung.

a) Geben Sie das exakte, zweiseitige 95 %-Vertrauensintervall für die wahre Differenz an, unter der Annahme, dass die Messfehler normal-verteilt sind.


In [36]:
n = 9
a = 0.95
ew = -403
sa = 3.127

norm.interval(alpha=a, loc=ew, scale=sa/np.sqrt(n))

(-405.04293579321893, -400.95706420678107)

b) Könnte die wahre Differenz −400.0 µg betragen? Entscheiden Sie aufgrund des
Resultats in Aufgabe a). (Kurze Begründung)

Aufgrund des Vertrauensintervalls würde die H0 verworfen werden

## Aufgabe 7.5
Eine Brücke soll aufgrund des höheren Verkehrsaufkommens renoviert werden. Im
Bau wurden damals Schrauben mit einer mittleren Festigkeit von 500 N/mm2 benutzt. Da dies für nicht mehr sicher genug gehalten wird, sollen diese nun durch
Schrauben mit einer mittleren Festigkeit von mehr als 500 N/mm2
ersetzt werden.
Um diesen Anforderungen gerecht zu werden, hat der alte Schraubenlieferant ein
neues Verfahren entwickelt. Zur Baustelle werden allerdings unbeschriftete Schrauben geliefert, aus denen nicht sofort hervorgeht, ob es sich um die alten 500er oder
um die neuen verbesserten Schrauben handelt.
Vor dem Verbau will der leitende Ingenieur zuerst sicherstellen, dass es sich um die
besseren Schrauben handelt. Um dies herauszufinden, werden einige der Schrauben
vermessen und ein statistischer Test durchgeführt. Je nach Ergebnis sollen die Schrauben verbaut oder zurückgeschickt werden.


Für den empirischen Mittelwert und empirische Varianz ergeben sich bei obiger Stichprobe x5 = 512 und s^2 x = 106.5.
Wir modellieren die Daten mit einer Normalverteilung, d. h. Xi
i.i.d. ∼ N (µ, σ
2
).

a) Stellen Sie die geeigneten Null- und Alternativhypothesen auf und begründen
Sie Ihre Wahl.

H0: u = 500
Ha: u > 500

b) Sie führen nun einen einseitigen t-Test zum Niveau α = 0.05 durch (unabhängig
von Ihrer obigen Antwort). Stellen Sie die Teststatistik T auf und berechnen
Sie deren Wert. Geben Sie die Verteilung der Teststatistik T unter H0 und den
Verwerfungsbereich des Tests an. Was ist der Testentscheid?


In [38]:
import numpy as np
from scipy.stats import norm, t
from pandas import Series
x = Series([520, 512, 499, 524, 505])
print(x.mean())
print(x.var())

512.0
106.5


In [39]:
a=0.95
n=x.size
ew=500
sa=x.std()
t.ppf(q=a, df=n-1, loc=ew, scale=sa/np.sqrt(n))

509.8388828578604

Verwerfungsbereich = [509.8, unendlich]

0 Hypothese wird verworfen --> schrauben sind stärker


c) Berechnen Sie ein (zweiseitiges) 95 %-Vertrauensintervall für µ.

In [42]:
x = Series([520, 512, 499, 524, 505])
n = x.size
t.interval(alpha=a, df=n-1, loc=x.mean(), scale=x.std()/np.sqrt(n))

(499.1861719188852, 524.8138280811148)

Wie würde das entsprechende Vertrauensintervall von c) aussehen, wenn wir
die Streuung als bekannt voraussetzen würden (mit dem gleichen Wert wie der
beobachtete)?

In [43]:
norm.interval(alpha=a, loc=x.mean(), scale=x.std()/np.sqrt(n))

(502.95438930304954, 521.0456106969505)